### PR TITLE
flirc_util: add initial add-on package

### PR DESCRIPTION
**Pull request by chewitt**
This PR creates a new add-on package for the `flirc_util` tool that allows flirc USB IR receiver configuration and firmware updates to be done from the SSH console as an alternative to Windows/macOS and desktop Linux GUI tools. It also adds a package for the hidapi dependency.

Test add-ons for ARMv7/ARMv8/Generic can be found here: https://chewitt.libreelec.tv/testing/addons/11.80.3/ .. it would be appreciated if folks with ARMv7 or x86_64 hardware could install them to test as these days all the hardware I have to hand is ARMv8 (which works fine). You don't need a flirc receiver to test, although that would be nice.

~NB: Draft PR pending on cross-compile improvements being merged upstream.~

### CODE REVIEW

**Comment by chewitt** [comment]
CCX changes are done upstream https://github.com/flirc/sdk/commits/master so ready for review/merge.

File: packages/addons/addon-depends/hidapi/package.mk [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2023-present Team LibreELEC (https://libreelec.tv)
+
+PKG_NAME="hidapi"
+PKG_VERSION="0.14.0"
+PKG_SHA256="a5714234abe6e1f53647dd8cba7d69f65f71c558b7896ed218864ffcf405bcbd"
+PKG_LICENSE="HIDAPI-orig"
```
Comment (CvH) [inline packages/addons/addon-depends/hidapi/package.mk:7]: The licence is clusterfu
```
1. The GNU General Public License, version 3.0, in LICENSE-gpl3.txt
2. A BSD-Style License, in LICENSE-bsd.txt.
3. The more liberal original HIDAPI license. LICENSE-orig.txt
```
https://github.com/libusb/hidapi/blob/master/LICENSE.txt so we would "need" 3 licences here -.-
not sure how we handle that

File: packages/addons/addon-depends/hidapi/package.mk [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (CvH) [inline packages/addons/addon-depends/hidapi/package.mk:1]:
```suggestion
# SPDX-License-Identifier: GPL-2.0-only
```

File: packages/addons/tools/flirc_util/package.mk [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# SPDX-License-Identifier: GPL-2.0
```
Comment (CvH) [inline packages/addons/tools/flirc_util/package.mk:1]:
```suggestion
# SPDX-License-Identifier: GPL-2.0-only
```

File: packages/addons/tools/flirc_util/source/default.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: GPL-2.0
```
Comment (CvH) [inline packages/addons/tools/flirc_util/source/default.py:1]:
```suggestion
# SPDX-License-Identifier: GPL-2.0-only
```

**Review by CvH — CHANGES_REQUESTED** [review]

File: packages/addons/addon-depends/hidapi/package.mk (hunk 1 above)
Comment (chewitt) [reply packages/addons/addon-depends/hidapi/package.mk:7]: This just means it's up to us to choose the license we use (from the three available) so I picked the original (most liberal) one. No need to do more.

**Pull request merged**